python/pyarrow/array.pxi

```diff
@@ -802,6 +802,12 @@ cdef class Array(_PandasConvertible):
         """
         return _pc().call_function('sum', [self])
 
+    def mode(self):
+        """
+        Compute the mode of valuesd in a numerical array.
+        """
+        return _pc().call_function('mode', [self])
```

Comment:
I am not sure we necessarily want to add it as a Array method as well. For a starter, it's also not a method on a numpy array (so convenient consistency with numpy is already not an argument). In general we don't expose all compute functions also as methods (but currently the line is not very clear ... )

Reply:
Agreed with @jorisvandenbossche .